v2/pkg/genruntime/admissions.go

```diff
@@ -26,3 +28,22 @@ type Defaulter interface {
 	// CustomDefault performs custom defaults that are run in addition to the code generated defaults.
 	CustomDefault()
 }
+
+// ValidateImmutableProperties function validates the update on immutable properties.
+func ValidateImmutableProperties(oldObj MetaObject, newObj MetaObject) error {
```

Comment:
I'm not sure "Immutable" is quite the right term for these properties, but I'm having trouble coming up with a reasonable alternative.